### ISSUE TITLE
Crashplan not running properly post-upgrade

### ISSUE THREAD

**Issue by Skiptar** [jlesage/docker-crashplan-pro#112]
Hiya. I found that I had a long list of "upgrade errors" so downloaded the latest version and recreated the container from the command line.

Since then I've had a bunch of issues that seem to have caused Crashplan to stop backing up successfully:
- Synchronisation to the 2 backup locations seemed to fail, causing it to think that it had to start from scratch
- It couldn't seem to see the files on the local drive (reporting 0M to back up)
- The client screen runs extremely slowly and sometimes hangs (device detail screen shows mostly blank)
- History shows that files to back up are being found, but there's no results for actually backing up
- The electron process is using a lot of memory
- The Java process is using a lot of CPU

Stopping and restarting the container seems to fix things temporarily (UI works and the server seems to be doing something) but seems to ultimately deliver a different error/issue after a while.

Any suggestions on how I can recover this?

I just pulled the very latest docker package. cp_verson showing as 6.7.2.

**Comment by Skiptar** [comment]
And BTW, Huzzah for this package! It's pretty awesome!!

**Comment by jlesage** [comment]
So on which system are you running the container?  What is your `docker run` command?  Could you provide the output of `docker inspect <name of your container>`?

**Comment by Skiptar** [comment]
Thanks for coming back so quickly. Here they are.

docker run -d --name=crashplan-pro -e USER_ID=0 -e GROUP_ID=0 \
        -e CRASHPLAN_SRV_MAX_MEM=4096M -p 5800:5800 -p 5900:5900 \
        -v /volume1/docker/appdata/crashplan-pro:/config:rw \
        -v /volume1/:/volume1:ro \
        -v /volumeUSB2/usbshare:/volumeUSB2/usbshare:rw \
        jlesage/crashplan-pro

[{
    "Architecture": "amd64",
    "Author": "",
    "Comment": "",
    "Config": {
        "AttachStderr": false,
        "AttachStdin": false,
        "AttachStdout": false,
        "Cmd": [
            "/init"
        ],
        "CpuShares": 0,
        "Cpuset": "",
        "Domainname": "",
        "Entrypoint": null,
        "Env": [
            "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
            "LANG=en_US.UTF-8",
            "S6_BEHAVIOUR_IF_STAGE2_FAILS=3",
            "S6_SERVICE_DEPS=1",
            "USER_ID=1000",
            "GROUP_ID=1000",
            "APP_NAME=CrashPlan for Small Business",
            "APP_USER=app",
            "XDG_DATA_HOME=/config/xdg/data",
            "XDG_CONFIG_HOME=/config/xdg/config",
            "XDG_CACHE_HOME=/config/xdg/cache",
            "DISPLAY=:0",
            "DISPLAY_WIDTH=1280",
            "DISPLAY_HEIGHT=768",
            "S6_WAIT_FOR_SERVICE_MAXTIME=10000",
            "KEEP_APP_RUNNING=1",
            "CRASHPLAN_DIR=/usr/local/crashplan",
            "JAVACOMMON=/usr/local/crashplan/jre/bin/java"
        ],
        "ExposedPorts": {
            "5800/tcp": {},
            "5900/tcp": {}
        },
        "Hostname": "",
        "Image": "sha256:49353730fe1c24e8fcf51bd7c078023c5e5b6f1ca00490d1b8b351ff728200be",
        "Labels": {
            "org.label-schema.description": "Docker container for CrashPlan PRO",
            "org.label-schema.name": "crashplan-pro",
            "org.label-schema.schema-version": "1.0",
            "org.label-schema.vcs-url": "https://github.com/jlesage/docker-crashplan-pro",
            "org.label-schema.version": "2.3.7"
        },
        "MacAddress": "",
        "Memory": 0,
        "MemorySwap": 0,
        "NetworkDisabled": false,
        "OnBuild": [],
        "OpenStdin": false,
        "PortSpecs": null,
        "StdinOnce": false,
        "Tty": false,
        "User": "",
        "Volumes": {
            "/config": {},
            "/storage": {}
        },
        "WorkingDir": "/tmp"
    },
    "Container": "70760d339dd79c3bcdbe9a489d98b877b78760a1523e1f4ed19f4411224f836f",
    "ContainerConfig": {
        "AttachStderr": false,
        "AttachStdin": false,
        "AttachStdout": false,
        "Cmd": [
            "/bin/sh",
            "-c",
            "#(nop) ",
            "LABEL org.label-schema.name=crashplan-pro org.label-schema.description=Docker container for CrashPlan PRO org.label-schema.version=2.3.7 org.label-schema.vcs-url=https://github.com/jlesage/docker-crashplan-pro org.label-schema.schema-version=1.0"
        ],
        "CpuShares": 0,
        "Cpuset": "",
        "Domainname": "",
        "Entrypoint": null,
        "Env": [
            "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
            "LANG=en_US.UTF-8",
            "S6_BEHAVIOUR_IF_STAGE2_FAILS=3",
            "S6_SERVICE_DEPS=1",
            "USER_ID=1000",
            "GROUP_ID=1000",
            "APP_NAME=CrashPlan for Small Business",
            "APP_USER=app",
            "XDG_DATA_HOME=/config/xdg/data",
            "XDG_CONFIG_HOME=/config/xdg/config",
            "XDG_CACHE_HOME=/config/xdg/cache",
            "DISPLAY=:0",
            "DISPLAY_WIDTH=1280",
            "DISPLAY_HEIGHT=768",
            "S6_WAIT_FOR_SERVICE_MAXTIME=10000",
            "KEEP_APP_RUNNING=1",
            "CRASHPLAN_DIR=/usr/local/crashplan",
            "JAVACOMMON=/usr/local/crashplan/jre/bin/java"
        ],
        "ExposedPorts": {
            "5800/tcp": {},
            "5900/tcp": {}
        },
        "Hostname": "70760d339dd7",
        "Image": "sha256:49353730fe1c24e8fcf51bd7c078023c5e5b6f1ca00490d1b8b351ff728200be",
        "Labels": {
            "org.label-schema.description": "Docker container for CrashPlan PRO",
            "org.label-schema.name": "crashplan-pro",
            "org.label-schema.schema-version": "1.0",
            "org.label-schema.vcs-url": "https://github.com/jlesage/docker-crashplan-pro",
            "org.label-schema.version": "2.3.7"
        },
        "MacAddress": "",
        "Memory": 0,
        "MemorySwap": 0,
        "NetworkDisabled": false,
        "OnBuild": [],
        "OpenStdin": false,
        "PortSpecs": null,
        "StdinOnce": false,
        "Tty": false,
        "User": "",
        "Volumes": {
            "/config": {},
            "/storage": {}
        },
        "WorkingDir": "/tmp"
    },
    "Created": "2018-06-14T09:59:24.262789729Z",
    "DockerVersion": "17.06.2-ee-6",
    "Id": "6a715ab35be6cfaa20cef906b97610eed20cee2972967ad7e1fb11d2d489ea49",
    "Os": "linux",
    "Parent": "6457cb4dc57b3e77bfed196ccc43c0abe0d1da270741696a10f27d8491638979",
    "Size": 0,
    "VirtualSize": 519533055
}
]

**Comment by Skiptar** [comment]
And I'm running on a Synology box running DSM 5.2.

Linux version 3.10.35 (root@ubuntu) (gcc version 4.7.3 20130102 (prerelease) (crosstool-NG 1.18.0) ) #1 SMP Sat Dec 12 17:01:14 MSK 2015

**Comment by jlesage** [comment]
How much memory you Synology NAS has?

**Comment by Skiptar** [comment]
It's got 2G of physical memory. The Crashplan support line told me to set Java to 4096 in spite of the physical limitation.

**Comment by jlesage** [comment]
2G of memory on Synology is apparently not enough to run CrashPlan...

See https://github.com/jlesage/docker-crashplan-pro/issues/82

**Comment by Skiptar** [comment]
Blast! 2G had been enough under the 4.0 client/server set up and had worked well with your docker container until this latest update. I wonder what changed...

Memory upgrade on order. Fingers crossed that it fixes the issues.

**Comment by jlesage** [comment]
> Memory upgrade on order. Fingers crossed that it fixes the issues.

Yes it should. :)

>  I wonder what changed...

I think the new UI is part of the problem... more resource consumed.
Or does the amount of data to backup increased?

Please re-open if you still have problems after the memory upgrade.

**Comment by Skiptar** [comment]
Hiya. Figured I'd post a quick follow-up...
I installed additional memory this morning taking the system from 2G to 6G. I removed the old container and restarted from the command line.

Crashplan immediately started behaving better: more responsive UI, more consistent outputs. It did a quick complete on the local external drive backup and seems to be in a good state; I can get a list of files to restore again!

Unfortunately it seems to have decided that I don't have anything backed up on the Crashplan Cloud and is now starting to back-up all 4TB from scratch. That seems odd as the Crashplan admin web site thinks I've got 3TB+ of data stored there. With luck it will figure that out and won't take the current estimate of 3 months to complete my backup...

Thanks for your guidance to the memory problem. Definitely seems that this latest Crashplan update was just too big for my 2G of memory to cope with. Shame it just started behaving oddly rather than telling me!
